scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -278,23 +278,40 @@ class Router(formatOps: FormatOps) {
           Split(Space, 0, ignoreIf = !canBeSpace),
           Split(afterCurlyNewlines, 1).withIndent(2, endOfFunction, Left)
         )
+
       case FormatToken(T.RightArrow(), right, _)
           if leftOwner.is[Term.Function] =>
-        val (endOfFunction, expiresOn) = functionExpire(
-          leftOwner.asInstanceOf[Term.Function]
-        )
-        val hasBlock =
-          nextNonComment(formatToken).right.isInstanceOf[T.LeftBrace]
-        val indent = // don't indent if the body is empty `{ x => }`
-          if (isEmptyFunctionBody(leftOwner) && !right.is[T.Comment]) 0
-          else 2
-        Seq(
-          Split(Space, 0, ignoreIf = isSingleLineComment(right))
-            .withPolicy(SingleLineBlock(endOfFunction)),
-          Split(Space, 0, ignoreIf = !hasBlock),
-          Split(Newline, 1 + nestedApplies(leftOwner), ignoreIf = hasBlock)
+        val lambda = leftOwner.asInstanceOf[Term.Function]
+        val (endOfFunction, expiresOn) = functionExpire(lambda)
+        val rightCommentedOut = isSingleLineComment(right)
+        val singleLineSplit =
+          Split(Space, 0, ignoreIf = rightCommentedOut)
+            .withPolicy(SingleLineBlock(endOfFunction))
+        def newlineSplit = {
+          val indent = // don't indent if the body is empty `{ x => }`
+            if (isEmptyFunctionBody(leftOwner) && !right.is[T.Comment]) 0
+            else 2
+          Split(Newline, 1 + nestedApplies(leftOwner))
             .withIndent(indent, endOfFunction, expiresOn)
-        )
+        }
+        val multiLineSplits =
```

Comment:
It would be helpful if you add some comments on the behavior of these conditional branches?